### PR TITLE
Add dxtbx

### PR DESCRIPTION
**Pull request by ndevenish**
Draft PR to test build since it doesn't seem to be working locally. An official (tagged) source will be used when ready to come out of draft.

- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Anthchirp** [comment]
For the record: happy to be a maintainer

File: recipes/dxtbx/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+{% set name = "dxtbx" %}
+{% set version = "3.8.2" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/cctbx/dxtbx/archive/refs/tags/v3.8.2.tar.gz
```
Comment (Anthchirp) [inline recipes/dxtbx/meta.yaml:9]: This should use the version variable

File: recipes/dxtbx/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    - dxtbx/tests
+    - dxtbx/conftest.py
+  requires:
+    - dials-data
+    - pytest
+    - pytest-mock
+  imports:
+    - dxtbx.ext
+  commands:
+    - pytest $SRC_DIR/dxtbx/tests --regression
```
Comment (Anthchirp) [inline recipes/dxtbx/meta.yaml:56]: Can we add a `pip check` here? (Needs `pip` as testing dependency ofc)

**Review by Anthchirp — APPROVED** [review]
Looks good judging from the recipe, haven't checked build output

**Comment by rjgildea** [comment]
Happy to be listed as a maintainer 👍

**Review by Anthchirp — APPROVED** [review]

**Pull request merged**